Shared/Samples/Edit feature attachments/EditFeatureAttachmentsView.Model.swift

```diff
@@ -0,0 +1,112 @@
+        let featureLayer = FeatureLayer(featureTable: ServiceFeatureTable(url: .featureServiceURL))
+        
+        init() {
+            map.addOperationalLayer(featureLayer)
+        }
+        
+        /// Selects the tapped feature.
+        /// - Parameter feature: The selected feature.
+        func selectFeature(_ feature: ArcGISFeature) async throws {
+            selectedFeature = feature
+            featureLayer.selectFeature(feature)
+            try await fetchAttachmentsAndUpdateFeature(feature: feature)
+        }
+        
+        /// Updates the callout text for the selected feature.
+        private func updateCalloutDetailsForSelectedFeature(feature: ArcGISFeature) {
+            let title = feature.attributes["typdamage"] as? String
+            calloutText = title ?? "Callout"
+            calloutDetailText = "Number of attachments: \(attachments.count)"
+        }
+        
+        /// Adds a new attachment with the given parameters and syncs this change with the server.
+        /// - Parameters:
+        ///   - name: The attachment name.
+        ///   - type: The attachments data type.
+        ///   - dataElement: The attachment data.
+        func addAttachment(named name: String, type: String, dataElement: Data) async throws {
+            if let selectedFeature,
+               let table = selectedFeature.table as? ServiceFeatureTable,
+               table.hasAttachments {
+                try await selectedFeature.addAttachment(
+                    named: "Attachment.png",
+                    contentType: "png",
+                    data: dataElement
+                )
+                _ = try await table.applyEdits()
+                try await fetchAttachmentsAndUpdateFeature(feature: selectedFeature)
+            }
+        }
+        
+        /// Deletes the specified attachment and syncs the changes with the server.
+        /// - Parameter attachment: The attachment to be deleted.
+        func deleteAttachment(_ attachment: Attachment) async throws {
+            if let selectedFeature,
+               let table = selectedFeature.table as? ServiceFeatureTable,
+               table.hasAttachments {
+                try await selectedFeature.deleteAttachment(attachment)
+                _ = try await table.applyEdits()
+                try await fetchAttachmentsAndUpdateFeature(feature: selectedFeature)
+            }
+        }
+        
+        /// Fetches attachments for the selected feature from the server.
+        private func fetchAndUpdateAttachments(feature: ArcGISFeature) async throws {
+            if let table = feature.table as? ServiceFeatureTable,
+               table.hasAttachments {
+                attachments = try await feature.attachments
+            }
+        }
+        
+        /// Fetches attachments from server and updates the selected feature's callout with the details.
+        private func fetchAttachmentsAndUpdateFeature(feature: ArcGISFeature) async throws {
```

Comment:
Method signature can be adjusted here as well, see the comment above.

Reply:
renamed  private func refreshAttachments(feature: ArcGISFeature)